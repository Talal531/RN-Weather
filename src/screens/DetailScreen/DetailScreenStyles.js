import {StyleSheet} from 'react-native';
import {color} from '../../theme';
import {moderateScale} from 'react-native-size-matters';

export const DetailStyle = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: color.palette.white,
    paddingHorizontal: moderateScale(15),
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: moderateScale(15),
    marginBottom: moderateScale(10),
  },
  menuImg: {
    height: moderateScale(10),
    width: moderateScale(20),
    resizeMode: 'stretch',
  },
  searchImg: {
    height: moderateScale(15),
    width: moderateScale(15),
    resizeMode: 'stretch',
  },
  content: {},
  countryBox: {flexDirection: 'row'},
  countryNameBox: {
    flex: 0.6,
  },
  countryImgBox: {
    flex: 0.4,
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
  },
  countryImg: {
    height: moderateScale(90),
    width: moderateScale(110),
    overflow: 'hidden',
    borderRadius: moderateScale(20),
  },
  title: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(20),
    fontWeight: '600',
  },
  subTitle: {
    color: color.palette.ThemeGray,
    fontSize: moderateScale(12),
    marginTop: moderateScale(5),
  },
  dayWeather: {
    flexDirection: 'row',
    height: moderateScale(93),
    borderRadius: moderateScale(20),
    backgroundColor: color.palette.ThemeCoralCandy,
    marginTop: moderateScale(40),
    marginBottom: moderateScale(45),
    paddingVertical: moderateScale(5),
    paddingHorizontal: moderateScale(15),
  },
  wetherDetail: {
    alignItems: 'center',
    justifyContent: 'center',
    height: moderateScale(80),
    borderRadius: moderateScale(20),
    flex: 1,
  },
  wetherTimeText: {
    color: color.palette.white,
  },
  wetherImg: {
    height: moderateScale(26),
    width: moderateScale(26),
    resizeMode: 'stretch',
    marginVertical: moderateScale(7),
  },
  wetherTemperature: {
    color: color.palette.white,
  },
  additionalInfoBox: {
    borderTopColor: color.palette.offWhite,
    borderTopWidth: moderateScale(1),
    borderBottomColor: color.palette.offWhite,
    borderBottomWidth: moderateScale(1),
    paddingVertical: moderateScale(18),
    marginBottom: moderateScale(15),
  },
  additionalTitle: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(18),
    fontWeight: '600',
    marginBottom: moderateScale(10),
  },
  additionalDetailBox: {
    flexDirection: 'row',
  },
  additionalDetail: {
    flex: 1,
  },
  additionalDetailSubTitle: {
    color: color.palette.ThemeGray,
    fontSize: moderateScale(14),
    marginBottom: moderateScale(5),
  },
  additionalDetailText: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(16),
    fontWeight: '600',
  },
  temperatureTitle: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(16),
    fontWeight: '600',
    marginBottom: moderateScale(10),
  },
});
