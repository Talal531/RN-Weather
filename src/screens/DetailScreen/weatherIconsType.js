import React, {Component, useState, useEffect} from 'react';
import {Image, View, Text} from 'react-native';
import {DetailStyle} from './DetailScreenStyles';
import moment from 'moment';
const weatherStatus = CurrentWether => {
  switch (CurrentWether) {
    case 'weather-mostly-cloud':
      return require('assets/imgs/weather_mostly_cloud_medium.png');
    case 'weather-clouds':
      return require('assets/imgs/weather-clouds_medium.png');
    case 'weather-cloud-thunder-rain':
      return require('assets/imgs/weather-cloud-thunder-rain_medium.png');
    case 'weather-sun':
      return require('assets/imgs/weather-sun_medium.png');
    default:
      return require('assets/imgs/weather-sun_medium.png');
  }
};

export function RenderByWeatherTypes(daily, index) {
  let {item} = daily.data;
  let src = weatherStatus('weather-sun');
  console.log('renderByWeatherTypes', index);
  return (
    <View
      style={[
        DetailStyle.wetherDetail,
        {backgroundColor: 'rgba(255, 255, 255, 0.1)'},
      ]}>
      <View style={DetailStyle.wetherDetail}>
        <Text style={DetailStyle.wetherTimeText}>
          {moment().format('hh:mm')}
        </Text>
        <Image style={DetailStyle.wetherImg} source={src} />
        <Text style={DetailStyle.wetherTemperature}>{`${item.temp.day}`}</Text>
      </View>
    </View>
  );
}
