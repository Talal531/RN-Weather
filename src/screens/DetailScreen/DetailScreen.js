import React, {Component, useState, useEffect, useCallback} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  TouchableOpacity,
  FlatList,
} from 'react-native';
import {color} from '../../theme';
import {moderateScale} from 'react-native-size-matters';
import {ScrollView} from 'react-native-gesture-handler';
import {useSelector, useDispatch} from 'react-redux';
import moment from 'moment';
import {_getWeatherByLocationHourly} from '../../services/weatherServices';
import {storeWeatherHourly} from '../../store/actions';
import {RenderByWeatherTypes} from './weatherIconsType';
import {DetailStyle} from './DetailScreenStyles';
export default function DetailScreen() {
  const [loading, setLoading] = useState(false);
  const weather = useSelector(state => state.weather);
  const location = useSelector(state => state.location);
  const weatherHourly = useSelector(state => state.weatherHourly);
  const dispatch = useDispatch();
  const [date, setDate] = useState(moment().format('MMM DD, YYYY'));

  useEffect(() => {
    async function getWeatherHourly() {
      let hourlyWeather = await _getWeatherByLocationHourly(location);
      console.log('_getWeatherByLocationHourly', hourlyWeather);
      setWeatherHourly(hourlyWeather);
    }

    getWeatherHourly();
  }, []);

  const setWeatherHourly = useCallback(
    data => dispatch(storeWeatherHourly(data)),
    [dispatch],
  );

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity onPress={() => {}}>
          <Image
            style={styles.menuImg}
            source={require('assets/imgs/menu_medium.png')}
          />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {}}>
          <Image
            style={styles.searchImg}
            source={require('assets/imgs/search_medium.png')}
          />
        </TouchableOpacity>
      </View>
      <ScrollView>
        <View style={styles.content}>
          <View style={styles.countryBox}>
            <View style={styles.countryNameBox}>
              <Text style={styles.title}>{` ${weather.name},`}</Text>
              <Text style={styles.title}>{` ${weather.sys.country},`}</Text>
              <Text style={styles.subTitle}>{date}</Text>
            </View>
            <View style={styles.countryImgBox}>
              <Image
                style={styles.countryImg}
                source={require('assets/imgs/bg.png')}
              />
            </View>
          </View>
          <View style={styles.dayWeather}>
            <FlatList
              data={weatherHourly ? weatherHourly.daily : []}
              numColumns={5}
              renderItem={(item, index) => {
                return <RenderByWeatherTypes data={item} index={index} />;
              }}
            />
          </View>
          <View style={styles.additionalInfoBox}>
            <Text style={styles.additionalTitle}>Additional Info</Text>
            <View style={styles.additionalDetailBox}>
              <View style={styles.additionalDetail}>
                <Text style={styles.additionalDetailSubTitle}>
                  Precipitation
                </Text>
                <Text style={styles.additionalDetailText}>3%</Text>
              </View>
              <View style={styles.additionalDetail}>
                <Text style={styles.additionalDetailSubTitle}>Humidity</Text>
                <Text style={styles.additionalDetailText}>{`${
                  weather.main.humidity
                } %`}</Text>
              </View>
              <View style={styles.additionalDetail}>
                <Text style={styles.additionalDetailSubTitle}>Windly</Text>
                <Text style={styles.additionalDetailText}>{`${
                  weather.wind.speed
                } km/h`}</Text>
              </View>
            </View>
          </View>
          <View style={styles.temperatureBox}>
            <Text style={styles.temperatureTitle}>Temperature</Text>
          </View>
        </View>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: color.palette.white,
    paddingHorizontal: moderateScale(15),
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: moderateScale(15),
    marginBottom: moderateScale(10),
  },
  menuImg: {
    height: moderateScale(10),
    width: moderateScale(20),
    resizeMode: 'stretch',
  },
  searchImg: {
    height: moderateScale(15),
    width: moderateScale(15),
    resizeMode: 'stretch',
  },
  content: {},
  countryBox: {flexDirection: 'row'},
  countryNameBox: {
    flex: 0.6,
  },
  countryImgBox: {
    flex: 0.4,
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
  },
  countryImg: {
    height: moderateScale(90),
    width: moderateScale(110),
    overflow: 'hidden',
    borderRadius: moderateScale(20),
  },
  title: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(20),
    fontWeight: '600',
  },
  subTitle: {
    color: color.palette.ThemeGray,
    fontSize: moderateScale(12),
    marginTop: moderateScale(5),
  },
  dayWeather: {
    flexDirection: 'row',
    height: moderateScale(93),
    borderRadius: moderateScale(20),
    backgroundColor: color.palette.ThemeCoralCandy,
    marginTop: moderateScale(40),
    marginBottom: moderateScale(45),
    paddingVertical: moderateScale(5),
    paddingHorizontal: moderateScale(15),
  },
  wetherDetail: {
    // alignItems: 'center',
    justifyContent: 'space-evenly',
    height: moderateScale(80),
    borderRadius: moderateScale(20),
    flex: 1,
  },
  wetherTimeText: {
    color: color.palette.white,
  },
  wetherImg: {
    height: moderateScale(26),
    width: moderateScale(26),
    resizeMode: 'stretch',
    marginVertical: moderateScale(7),
  },
  wetherTemperature: {
    color: color.palette.white,
  },
  additionalInfoBox: {
    borderTopColor: color.palette.offWhite,
    borderTopWidth: moderateScale(1),
    borderBottomColor: color.palette.offWhite,
    borderBottomWidth: moderateScale(1),
    paddingVertical: moderateScale(18),
    marginBottom: moderateScale(15),
  },
  additionalTitle: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(18),
    fontWeight: '600',
    marginBottom: moderateScale(10),
  },
  additionalDetailBox: {
    flexDirection: 'row',
  },
  additionalDetail: {
    flex: 1,
  },
  additionalDetailSubTitle: {
    color: color.palette.ThemeGray,
    fontSize: moderateScale(14),
    marginBottom: moderateScale(5),
  },
  additionalDetailText: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(16),
    fontWeight: '600',
  },
  temperatureTitle: {
    color: color.palette.ThemeCoralCandy,
    fontSize: moderateScale(16),
    fontWeight: '600',
    marginBottom: moderateScale(10),
  },
});
