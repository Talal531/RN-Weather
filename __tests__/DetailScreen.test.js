import React from 'react';

import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import DetailScreen from '../src/screens/DetailScreen/DetailScreen';

import renderer from 'react-test-renderer';

const middlewares = [];
const mockStore = configureStore(middlewares);

const INITIAL_STATE = {
  weather: {
    name: '',
    sys: {
      country: '',
    },
    main: {
      humidity: '',
    },
    wind: {
      speed: '',
    },
  },
};

const updateLocation = () => ({
  type: 'UPDATE_LOCATION',
  payload: INITIAL_STATE,
});

describe('Detail Screen', () => {
  const store = mockStore(INITIAL_STATE);

  it('should renders correctly', async () => {
    const tree = renderer.create(
      <Provider store={store}>
        <DetailScreen />
      </Provider>,
    );
    const json = tree.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('dispatch update location', () => {
    store.dispatch(updateLocation());

    const actions = store.getActions();

    const expectedPayload = {type: 'UPDATE_LOCATION', payload: INITIAL_STATE};
    expect(actions).toEqual([expectedPayload]);
  });
});
