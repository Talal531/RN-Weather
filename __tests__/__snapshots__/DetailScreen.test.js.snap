// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Screen should renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingHorizontal": 23.57142857142857,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15.714285714285714,
        "paddingVertical": 23.57142857142857,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/imgs/menu_medium.png",
          }
        }
        style={
          Object {
            "height": 15.714285714285714,
            "resizeMode": "stretch",
            "width": 31.428571428571427,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/imgs/search_medium.png",
          }
        }
        style={
          Object {
            "height": 23.57142857142857,
            "resizeMode": "stretch",
            "width": 23.57142857142857,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#013239",
                  "fontSize": 31.428571428571427,
                  "fontWeight": "600",
                }
              }
            >
               ,
            </Text>
            <Text
              style={
                Object {
                  "color": "#013239",
                  "fontSize": 31.428571428571427,
                  "fontWeight": "600",
                }
              }
            >
               ,
            </Text>
            <Text
              style={
                Object {
                  "color": "#8d8d8d",
                  "fontSize": 18.857142857142858,
                  "marginTop": 7.857142857142857,
                }
              }
            >
              Jun 10, 2020
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.4,
                "justifyContent": "flex-end",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/imgs/bg.png",
                }
              }
              style={
                Object {
                  "borderRadius": 31.428571428571427,
                  "height": 141.42857142857144,
                  "overflow": "hidden",
                  "width": 172.85714285714283,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#013239",
              "borderRadius": 31.428571428571427,
              "flexDirection": "row",
              "height": 146.14285714285714,
              "marginBottom": 70.71428571428572,
              "marginTop": 62.857142857142854,
              "paddingHorizontal": 23.57142857142857,
              "paddingVertical": 7.857142857142857,
            }
          }
        >
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={5}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#e6e6e6",
              "borderBottomWidth": 1.5714285714285714,
              "borderTopColor": "#e6e6e6",
              "borderTopWidth": 1.5714285714285714,
              "marginBottom": 23.57142857142857,
              "paddingVertical": 28.285714285714285,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#013239",
                "fontSize": 28.285714285714285,
                "fontWeight": "600",
                "marginBottom": 15.714285714285714,
              }
            }
          >
            Additional Info
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#8d8d8d",
                    "fontSize": 22,
                    "marginBottom": 7.857142857142857,
                  }
                }
              >
                Precipitation
              </Text>
              <Text
                style={
                  Object {
                    "color": "#013239",
                    "fontSize": 25.142857142857142,
                    "fontWeight": "600",
                  }
                }
              >
                3%
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#8d8d8d",
                    "fontSize": 22,
                    "marginBottom": 7.857142857142857,
                  }
                }
              >
                Humidity
              </Text>
              <Text
                style={
                  Object {
                    "color": "#013239",
                    "fontSize": 25.142857142857142,
                    "fontWeight": "600",
                  }
                }
              >
                 %
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#8d8d8d",
                    "fontSize": 22,
                    "marginBottom": 7.857142857142857,
                  }
                }
              >
                Windly
              </Text>
              <Text
                style={
                  Object {
                    "color": "#013239",
                    "fontSize": 25.142857142857142,
                    "fontWeight": "600",
                  }
                }
              >
                 km/h
              </Text>
            </View>
          </View>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#013239",
                "fontSize": 25.142857142857142,
                "fontWeight": "600",
                "marginBottom": 15.714285714285714,
              }
            }
          >
            Temperature
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
